src/main/java/edu/greenblitz/robotName/commands/swerve/RotateToAngle.java

```diff
@@ -4,32 +4,40 @@
 import edu.greenblitz.robotName.subsystems.swerve.Chassis.SwerveChassis;
 import edu.wpi.first.math.geometry.Rotation2d;
 
-public class RotateToAngle extends SwerveCommand{
-    private Rotation2d angleSetPoint;
+import java.util.function.Supplier;
 
-    public RotateToAngle(Rotation2d angleSetPoint){
+public class RotateToAngle extends SwerveCommand {
+
+    private Supplier<Rotation2d> angleSetPoint;
+
+    public RotateToAngle(Rotation2d angleSetPoint) {
+        super();
+        this.angleSetPoint = () -> angleSetPoint;
+    }
+
+    public RotateToAngle(Supplier<Rotation2d> angleSetPoint) {
         super();
         this.angleSetPoint = angleSetPoint;
     }
 
     @Override
-    public void initialize(){
-        ChassisConstants.ROTATION_PID_CONTROLLER.setSetpoint(angleSetPoint.getRadians());
+    public void initialize() {
+        System.out.println("rotate init");
+        ChassisConstants.ROTATION_PID_CONTROLLER.setSetpoint(angleSetPoint.get().getRadians());
```

Comment:
Is this for simulation or something? It looks weird to change the setpoint directly from the command

Reply:
At least use a “setGoal” function in the subsystem or something

Reply:
It doesn't have anything to do with the simulation. This is just how Noam told me to write it. I can move it to a setGoalAngle function in the subsystem if you'd like.

Reply:
Yes please